BUILDING.md

```diff
@@ -53,66 +29,169 @@ following packages are installed:
+be overridden by setting `CMAKE_INSTALL_PREFIX` prior to build.
+
+By default, OpenAssetIO builds as a shared library. This can be
+overriden by setting `BUILD_SHARED_LIBS` prior to build.
+If shared library builds are disabled, then the core library will
+be built and installed as a static library, and statically linked into
+the Python/C modules.
+
+In installing the python module, OpenAssetIO creates a valid python
+install tree, placed at `build/dist/{python-version}/site-packages`.
+This generated install tree can then be used directly (e.g. by extending
 `PYTHONPATH` and/or `LD_LIBRARY_PATH`), or as the source for a further
 packaging step.
 
-Note that the `openassetio` Python package will not function unless the
-binary artifacts are also made available to your Python environment.
+> **Note**
+>
+> When built this way, the `openassetio` Python package will not
+> function unless the binary artifacts are also made available to your
+> Python environment.
+>
+
+### Building via pip
+
+OpenAssetIO can alternately be built and installed via pip, which may be
+preferable to the exclusively python focused developer.
+Check out the repository, and from the root, run :
+
+```shell
+pip install .
+```
+
+This will automatically build and install the Python extension module
+into your python environment, along with the Python sources. After this,
+you should be able to simply `import openassetio` from your python shell.
+
+> **Note**
+>
+> `pip install .` is not a substantially different operation to
+> calling CMake directly as described above, (and as such, all the same
+> dependency requirements apply).
+>
+> However, be aware that once the C++
+> library is built, it is statically, rather than dynamically, linked
+> into the python libraries, and is not usable as a standalone
+> OpenAssetIO C++ library.
+>
+> This means that this method of building is only appropriate for the
+> exclusively python focused developer. Anyone wishing to write
+> non-python hosts or managers should build via the CMake method
+> described above.
+
+For users that wish to modify the Python sources and test their changes
+without reinstalling, [editable installs](https://pip.pypa.io/en/stable/topics/local-project-installs/#editable-installs)
+are also supported
+
+```shell
+pip install --editable .
+```
+
+## CMake Options
+
+OpenAssetIO provides several custom CMake options to control the
+behaviour of the build.
+
+| Option      | Description | Default |
+| ----------- | ----------- | ------- |
+| `BUILD_SHARED_LIBS` | `ON` to build shared libraries. `OFF` to build static libraries. | `ON` |
+| `OPENASSETIO_ENABLE_PYTHON` | Additionally build python bindings | `ON` |
+| `OPENASSETIO_ENABLE_C` | Additionally build C bindings | `ON` |
+| `OPENASSETIO_ENABLE_TESTS` | Additionally build tests | `OFF` |
+| `OPENASSETIO_ENABLE_PYTHON_TEST_VENV` | Automatically create environment when running tests | `ON` if `OPENASSETIO_ENABLE_TESTS` is enabled, `OFF` otherwise. |
```

Comment:
[consider] It's technically true, but reading it here makes me think we shouldn't have bothered with a `cmake_dependent_option`, since it's already implicitly dependent.
```suggestion
| `OPENASSETIO_ENABLE_PYTHON_TEST_VENV` | Automatically create environment when running tests | `ON` |
```